redaxo/src/core/lib/console/setup/run.php

```diff
@@ -0,0 +1,542 @@
+                if (true !== $msg = $passwordPolicy->check($password)) {
+                    throw new InvalidArgumentException($msg);
+                }
+
+                return $password;
+            };
+
+            $pwQuestion = new Question('Password');
+            $pwQuestion->setHidden(true);
+            $pwQuestion->setValidator($pwValidator);
+            $password = $this->getOptionOrAsk(
+                $pwQuestion,
+                'admin-password',
+                null,
+                'Setting admin password: *secret*',
+                $pwValidator
+            );
+
+            $user = rex_sql::factory();
+            $user->setTable(rex::getTablePrefix() . 'user');
+            $user->setValue('login', $login);
+            $user->setValue('password', rex_backend_login::passwordHash($password));
+            $user->setValue('admin', 1);
+            $user->addGlobalCreateFields('console');
+            $user->addGlobalUpdateFields('console');
+            $user->setValue('status', '1');
+            $user->insert();
+
+            $io->success(sprintf('User "%s" successfully created.', $login));
+        } else {
+            $io->success('No additional admin user created');
+        }
+
+        // ---------------------------------- last step. save config
+
+        $config['setup'] = false;
+        if (!rex_file::putConfig($configFile, $config)) {
+            $io->error('Writing to config.yml failed.');
+            return 1;
+        }
+        rex_file::delete(rex_path::coreCache('config.yml.cache'));
+
+        $io->success('Congratulations! REDAXO has successfully been installed.');
+        return 0;
+    }
+
+    /**
+     * @return bool|string false|utf8|utf8mb4
+     */
+    private function getDbCharset()
+    {
+        if ($charset = $this->input->getOption('db-charset')) {
+            if (!in_array($charset, ['utf8', 'utf8mb4'])) {
+                throw new InvalidArgumentException('unknown database charset "'.$charset.'" specified');
+            }
+            if ('utf8mb4' === $charset && !rex_setup_importer::supportsUtf8mb4()) {
+                $sql = rex_sql::factory();
+                throw new InvalidArgumentException('Your database doesn\'t support utf8mb4. It requires at least MySQL 5.7.7 or MariaDB 10.2. You are using '.$sql->getDbType(). ' '.$sql->getDbVersion());
+            }
+            $this->io->success('Using database charset "'.$charset.'"');
+            return $charset;
+        }
+
+        if (!$this->input->isInteractive()) {
+            throw new InvalidArgumentException('Required option "db-charset" is missing');
+        }
+
+        if (!rex_setup_importer::supportsUtf8mb4()) {
+            $sql = rex_sql::factory();
+            $this->io->writeln('Your database doesn\'t support utf8mb4. It requires at least MySQL 5.7.7 or MariaDB 10.2. You are using '.$sql->getDbType(). ' '.$sql->getDbVersion());
```

Comment:
Weiß nicht so recht, wie man es besser formuliert, ggf. so?

```suggestion
            $this->io->writeln('The utf8mb4 charset in REDAXO requires at least MySQL 5.7.7 or MariaDB 10.2. You are using '.$sql->getDbType(). ' '.$sql->getDbVersion());
```

Reply:
besser. Denn MariaDB 10.1 liefert ja bereits einen Support, anscheinend aber nicht den richtigen?

Reply:
@skerbis siehe https://stackoverflow.com/a/43424427/3837585

Reply:
ok. danke